menu.py

```diff
@@ -49,8 +51,58 @@ def showMenu():
 ############################################################
 
 def generatePasswordPrompt():
-    #TODO
-    print("todo")
+    lc = getUserInput("Include lowercase? (y/n/E)")
+    uc = getUserInput("Incldue Uppercase? (y/n/E)")
+    dig = getUserInput("Include numbers? (y/n/E)")
+    punc = getUserInput("Include special characters? (y/n/E)")
+    spc = getUserInput("Include spaces? (y/n/E)")
+
+    if lc=='n' and uc=='n' and dig=='n' and punc=='n' and spc=='n':
+        print("No character set chosen")
+        return ""
+
+    siz = getUserInput("Password length")
+
+    if not siz=='' and not siz.isdecimal():
+        print("not a number")
+        return ""
+
+
+
+    size = int(siz) if siz else 30
+
+    if size < 5:
+        print("Minimum length is 5")
+        return ""
+
+    charlist = string.ascii_lowercase if not lc == 'n' else ''
+    charlist += string.ascii_uppercase if not uc =='n' else ''
+    charlist += string.digits if not dig == 'n' else ''
```

Comment:
Should we check for 'y' or 'E' here? Right now, it looks like anything that's not 'n' will get registered as 'E'. Although, maybe that's for the best. That way users can just hit enter through the prompts, and it'll register as either.

Asked and answered I guess.